spec/index.bs

```diff
@@ -2028,13 +2028,48 @@ dictionary GPUProgrammableStageDescriptor {
 };
 </script>
 
+<script type=idl>
+enum GPUCompilationMessageType {
+    'error',
+    'warning',
+    'info'
+};
+enum GPUProgrammableStageType {
+    'vertex',
+    'fragment',
+    'compute'
+};
+
+[Serializable]
+interface GPUCompilationMessage {
+    GPUCompilationMessageType type;
+    GPUProgrammableStageType stage;
+    unsigned long long lineNum;
+    unsigned long long linePos;
+    DOMString? mappedFileName;
+    unsigned long long mappedLineNum;
+    unsigned long long mappedLinePos;
+};
```

Comment:
Is there a message string as well?  I'm probably missing something, but I can't see anything that says what the actual error or warning might be.

Reply:
Haha, oops!